### PR TITLE
New update

### PR DESCRIPTION
**Pull request by 0Tron**
# Submitting Week 3 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Career Readiness
- [ ] Coding Challenges
- [ ] Interview Questions
- [ ] React Debugging

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: career-readiness.md [hunk 1]
```diff
@@ -8,15 +8,15 @@ Creating a technical portfolio website is a big undertaking. Portfolios are almo
 
 Wire frames are used to outline the basic structure of your site. Wire frames should be vague enough to give you flexibility in your development process but structured enough to ensure all appropriate elements are present. 
 
-Submit the link to your wire frames:
+Submit the link to your wire frames: https://www.figma.com/file/G3OBH0QKplDhUZNDTe2hlS/Figma-basics?type=design&node-id=505%3A12&mode=design&t=ZTXRPcLFSxGuUYk9-1
```
Comment (sjproctor) [inline career-readiness.md:11]: Good start!

File: code-challenges.test.js [hunk 2]
```diff
@@ -16,19 +16,50 @@
...
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
-
+const fibonacciLength2 = 10
 // b) Create the function that makes the test pass.
+describe("fibonacciLength2",() => {
+  it ("returns an array containing the Fibonacci sequence. The length of the array is determined by the argument of the function", () => {
+    expect(fibonnaci(fibonacciLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+  })
+})
+// i was able to get the test statements easily and the else if statements.
```
Comment (sjproctor) [inline code-challenges.test.js:32]: Remember that you only need one test per function. You can have as many expect statements as you need. Also ensure you put your test variables inside the scope of your test.

File: code-challenges.test.js [hunk 3]
```diff
@@ -16,19 +16,50 @@
...
+function fibonnaci(length) {
+  if (length <= 0) {
+    // if the length comes back as 0 return an empty array
+    return []; //[] represent an empty array
+  } else if (length === 1) { //if the length comes back as stricktly equal to 1 
+    return [1]; // return an array with the first fib
+  } else if (length === 2) { // if the length comes back as stricktly equal to 2
+  return [1, 1]; // return the first two fib
+}
+// This is where i was completely stuck, i knew that we had to use to variables in order for us to achieve the next number but i just didnt know how to code it.
```
Comment (sjproctor) [inline code-challenges.test.js:43]: It looks like you spent a long time on the edge cases that weren't required in the prompt. Fibonacci is a tough one because it takes some time to understand the logic. The next number is the sum of the previous two numbers. In an array, that would be the previous two values.

File: code-challenges.test.js [hunk 4]
```diff
@@ -16,19 +16,50 @@
...
+
+
+
+const sequenceOrder = [1, 1]; // Initialize with the first two
+for (let i = 2; i < length; i++) { // This will help us generate the following fibonacci sequence; i had to use help for this as i was stuck for a few hours
+  sequenceOrder.push(sequenceOrder[i - 1] + sequenceOrder[i - 2]);
+}
+  return sequenceOrder;
+}
+
```
Comment (sjproctor) [inline code-challenges.test.js:53]: This is getting closer. Just missing the function to wrap your code logic.

File: code-challenges.test.js [hunk 5]
```diff
@@ -48,9 +87,26 @@ const studyMinutesWeek2 = {
...
-// Pseudo code:
+function sortObjectValues(obj) {
+  // The function take the object and extracts its values
+  const values = Object.values(obj);
+  //it returns those values in ascending order with the .sort method
+  return values.sort((a, b) => a - b);
+}
+
+module.exports = { sortObjectValues };
+
```
Comment (sjproctor) [inline code-challenges.test.js:112]: I am curious what this line is doing for you`module.exports = { sortObjectValues };`?

File: react-debugging.html [hunk 6]
```diff
@@ -81,9 +84,14 @@ <h1>Draw a Card</h1>
               )
             })}
           </div>
+        </>
         )
       }
 
+      // The bugs i fixed were onClick (one of them was lowercase), the other first one was missing {drawCard} & there was no fragment components
+      // I used the console to guide me.
```
Comment (sjproctor) [inline react-debugging.html:92]: Nice job. Using the console is the exact purpose of this exercise.

**Review by sjproctor — APPROVED** [review]
Your React debugging looks great. The rest of your assessment feels very unorganized. I think more attention to detail will be helpful for you. Focus on quality over quantity and make sure your tests are structured correctly! This is showing a novice level of skill acquisition.
